core/src/main/java/com/datastrato/gravitino/lock/TreeLockNode.java

```diff
@@ -0,0 +1,170 @@
+/*
+ * Copyright 2024 Datastrato Pvt Ltd.
+ * This software is licensed under the Apache License version 2.
+ */
+
+package com.datastrato.gravitino.lock;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.collect.Lists;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.concurrent.locks.ReentrantReadWriteLock;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * TreeLockNode is a node in the tree lock; all tree lock nodes will be assembled to a tree
+ * structure, which corresponds to the resource path like name identifier space.
+ *
+ * <p>Each node will have a read-write lock to protect the node. The node will also have a map to
+ * store the children. For more, please refer to {@link TreeLock}.
+ */
+public class TreeLockNode {
+  public static final Logger LOG = LoggerFactory.getLogger(TreeLockNode.class);
+  private final String name;
+  private final ReentrantReadWriteLock readWriteLock;
+  @VisibleForTesting final Map<String, TreeLockNode> childMap;
+  private final LockManager lockManager;
+
```

Comment:
I suggest recording lock information, including the locked timestamp and the locked thread ID, and waiting threads ID for debug.

If the thread call lock function. if  the lock already locked for a long time, we should throw an exception to the user. The exception message should include the lock name, the ID of the thread that holds the lock, and the duration of the lock. This can prevent all threads from hanging on the lock and assist us in debugging the problems